### DISCUSSION TITLE
Whitelist/Restrict API requests to specified host/domain names

### DISCUSSION THREAD

**Discussion by brandonlenz** [vercel/next.js#44670] [Help]
Does NextJS have an equivalent to [Django's ALLOWED_HOSTS](https://docs.djangoproject.com/en/4.1/ref/settings/#allowed-hosts) for domains that the NextJS app is allowed to serve?

I see [something close to what I'm looking for specific to images](https://nextjs.org/docs/api-reference/next/image#remote-patterns), but I'm unsure if there's an equivalent for all requests to the NextJS server. 

Is this possible to do at the application level with NextJS?

**Comment by wwwdepot** [comment]
Use middleware to check request headers
